src/components/blocklot.js

```diff
@@ -0,0 +1,435 @@
+                setValue(map[prop], value)
+              }
+            }
+
+            function initData () {
+              // initialize data if the address has changed
+              var data = context.data.blocklot
+              if (data.selectProjectAddress !== data.projectAddress) {
+                updateAddressFields({})
+                if (data.selectProjectAddress) {
+                  getAddressItems()
+                } else {
+                  context.instance.triggerChange()
+                }
+              }
+            }
+            function setValue (key, value) {
+              // set data value and dom for a specific component
+              var comp = context.utils.getComponent(context.form.components[0].components, key)
+              var el = document.querySelector('#' + comp.id + ' input')
+              if (el) {
+                el.value = value
+                context.data.blocklot[key] = value
+              } else {
+                console.log('Cannot setValue of component ' + key)
+              }
+            }
+          },
+          key: 'projectAddressData',
+          type: 'hidden',
+          input: true,
+          tableView: false,
+          hideOnChildrenHidden: false
+        },
+        {
+          label: 'Project Address Data Items',
+          persistent: false,
+          customDefaultValue: '',
+          key: 'projectAddressDataItems',
+          type: 'hidden',
+          input: true,
+          tableView: false
+        },
+        {
+          label: 'Columns',
+          columns: [
+            {
+              components: [
+                {
+                  label: 'Block number',
+                  disabled: true,
+                  tableView: true,
+                  key: 'lookupBlockNumber',
+                  type: 'textfield',
+                  input: true,
+                  hideOnChildrenHidden: false
+                }
+              ],
+              width: 6,
+              offset: 0,
+              push: 0,
+              pull: 0,
+              size: 'md'
+            },
+            {
+              components: [
+                {
+                  label: 'Lot number',
+                  disabled: true,
+                  tableView: true,
+                  redrawOn: 'projectAddressData',
+                  customDefaultValue: (context) => {
+                    var data = context.data.blocklot
+                    if (data.projectAddressDataItems) {
+                      var dataItems = JSON.parse(data.projectAddressDataItems)
+                      if (dataItems.length) {
+                        return dataItems[0].lot
+                      }
+                    }
+                  },
+                  key: 'lookupLotNumber',
+                  customConditional: (context) => {
+                    return context.form.components[0].conditionalLots(context, lots => lots.length === 1, true)
+                  },
```

Comment:
I add `conditionalLots` function within the component but don't know a good way to reference in `customConditional`, so this is a bit clunky. @shawnbot let me know your thoughts on it.